src/main/java/graphql/schema/GraphQLArgument.java

```diff
@@ -56,10 +62,26 @@ public GraphQLInputType getType() {
         return type;
     }
 
+    /**
+     * An argument has a default value when it represents the logical argument structure that a {@link graphql.schema.GraphQLFieldDefinition}
+     * can have and it can also have a default value when used in a schema definition language (SDL) where the
+     * default value comes via the directive definition.
+     *
+     * @return the default value of an argument
+     */
     public Object getDefaultValue() {
         return defaultValue;
     }
 
+    /**
+     * An argument ONLY has a value when its used in a schema definition language (SDL) context as the arguments to SDL directives.  The method
+     * should not be called in a query context, but rather the AST / variables map should be used to obtain an arguments value.
+     *
+     * @return the argument value
+     */
+    public Object getValue() {
```

Comment:
maybe we could add some flag to make it more clear if this argument is for a directive or field?

Reply:
I disagree.  I don't think the mental cost is worth the benefit.

Reply:
ok